aptos-move/framework/aptos-framework/sources/Token.move

```diff
@@ -63,19 +63,33 @@ module AptosFramework::Token {
         amount: u64,
     }
 
+    /// create collection event with creator address and collection name
+    struct CreateCollectionEvent has drop, store {
+        creator: address,
+        collection_name: ASCII::String,
+    }
+
+    /// token creation event id of token created
+    struct CreateTokenEvent has drop, store {
+        id: TokenId,
+        max: Option<u64>,
```

Comment:
Since TokenData is store-able, may be we can include TokenData here in this event for completeness. It can save clients from having to do a second request to fetch the token data when they see this event.

Reply:
TokenData don't change with each new event. I think the client can save a token data for each token without needing to request to fetch token data with every new event. This can also save storage since we don't have to save duplicated data many times.

Reply:
Isn't this CreateTokenEvent for when a token is created though? So TokenData should be unique per event, shouldn't it? There's a map from TokenId to TokenData so it looks like each token has its own token data.

Reply:
you are right the token data is unique per createToken event.  I think one question I don't know is if clients rely on the events to get all the data they interested or they call some indexer API to get the data? If it is the latter one, we may not need to have all the data in events?

Reply:
TokenData should be all the data the user is interested in in this case. It's still early and hard to tell from an ecosystem perspective, but from my past experience building dapps on other chains, it makes lives a lot easier if the events contain all or most of the information needed, this doesn't just save one call to the JSON RPC api but can make building custom indexing pipelines/backends much easier (less data-specific). In this particular case for this token creation event, TokenData should be all the information required.

Reply:
good discussion. the token data is not derived data and will be save user effort down the line. will add the token data to the event.

Reply:
> huge 

what is "huge"?

Reply:
I totally agree we should include the TokenData in the event.
I think in principle the events are the preferred interface for parts of a dApp external to the Move contracts to programmatically interact with the chain (compared to reading directly the state). 
* event schemas are designed to be friendly for reading by such parts of the dApp; state is designed to match access patterns of the Move contracts
* historical events are stored longer than historical state; basically consider only the "current"/latest state is available for querying.
* An event stream can more easily represent the changes happened on concerned part of the chain. 

> what is "huge"?
I don't really care too much about that. Just like the granularity of state items should match the access pattern of them by Move contracts, we may want to make sure we are not getting large events for only a small part of it that's of concern. We also don't want multiple events to be always got together but separately. Of course we should make sure we charge event emissions relative to their sizes, but generally events are cheaper to store on the DB size.